x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorPrivileges.java

```diff
@@ -78,6 +84,12 @@ public ElasticsearchSecurityException check(String action, TransportRequest requ
             return null;
         }
 
+        public void maybeInterceptRequest(ThreadContext threadContext, TransportRequest request) {
+            if (request instanceof RestoreSnapshotRequest) {
+                ((RestoreSnapshotRequest) request).skipOperatorOnly(shouldProcess());
```

Comment:
The code always sets the value of `skipOperatorOnly` because:
1. Similar to other authorization decision, the value is computed and local to every node
2. The value is always set properly so that a transport client cannot override this field (it is not exposed to Rest)